presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -152,7 +152,7 @@ public ElasticsearchTableHandle getTableHandle(ConnectorSession session, SchemaT
                 }
             }
 
-            if (listTables(session, Optional.of(schemaName)).contains(new SchemaTableName(schemaName, table))) {
+            if (client.indexExists(table) && !client.getIndexMetadata(table).getSchema().getFields().isEmpty()) {
```

Comment:
I the `client.getIndexMetadata(table).getSchema().getFields().isEmpty()` case covered by tests?

Reply:
Yes